src/namespace.ts

```diff
@@ -6,64 +6,39 @@ import * as fs from 'fs';
 
 export class Namespace
 {
-    private composerFile: string;
-    private namespaces: Array<Object> = new Array<Object>();
-    private namespaceCache: object = {};
-
-    constructor() {
-        this.composerFile = vscode.workspace.rootPath + path.sep + 'composer.json';
-
-        fs.watchFile(this.composerFile, () => {
-            console.log('composer.json change detected');
-            this.loadNamespaces();
-        });
-
-        this.loadNamespaces();
-    }
-
-    private loadNamespaces()
-    {
-        this.namespaces = new Array<Object>();
-        this.namespaceCache = {};
-
-        if (fs.existsSync(this.composerFile)) {
-            let buffer: Buffer = fs.readFileSync(this.composerFile);
-            let data = JSON.parse(buffer.toString());
-
-            if (data.hasOwnProperty('autoload') && data.autoload.hasOwnProperty('psr-4')) {
-                this.addNamespaces(data.autoload['psr-4']);
-            }
-            
-            if (data.hasOwnProperty('autoload-dev') && data['autoload-dev'].hasOwnProperty('psr-4')) {
-                this.addNamespaces(data['autoload-dev']['psr-4']);
-            }
-        }
-    }
+    private namespaces: object = {};
+    private knownComposerFiles: object = {};
 
     public async getNamespace(filePath: string, type: string)
     {
-        let relativePath = path.relative(vscode.workspace.rootPath, filePath);
-        let folder = path.dirname(relativePath);
+        let composerFile = this.knowsComposerFileForPath(filePath);
```

Comment:
method name suggestion: `resolveComposerFileForPath`, WDYT?

Reply:
So would lead to other kind of renames if you like this one, like:
- `knownComposerFiles => resolvedComposerFiles`
- `addComposerFile() => registerComposerFile()`

As you build kind of composer files registry on the fly, pretty like a DNS system where registration/resolution are the main operations you can perform.